fe/fe-core/src/main/java/com/starrocks/sql/optimizer/MaterializedViewOptimizer.java

```diff
@@ -125,16 +133,33 @@ private List<Range<PartitionKey>> getLatestPartitionRangeForTable(OlapTable part
         return latestBaseTableRanges;
     }
 
-    private List<Range<PartitionKey>> getLatestPartitionRange(OlapTable table, Set<String> modifiedPartitionNames) {
+    private List<Range<PartitionKey>> getLatestPartitionRangeForLocalTable(OlapTable partitionTable,
+                                                                           Set<String> modifiedPartitionNames) {
         // partitions that will be excluded
         Set<Long> filteredIds = Sets.newHashSet();
-        for (Partition p : table.getPartitions()) {
+        for (Partition p : partitionTable.getPartitions()) {
             if (modifiedPartitionNames.contains(p.getName()) || !p.hasData()) {
                 filteredIds.add(p.getId());
             }
         }
-        RangePartitionInfo rangePartitionInfo = (RangePartitionInfo) table.getPartitionInfo();
-        List<Range<PartitionKey>> latestPartitionRanges = rangePartitionInfo.getRangeList(filteredIds, false);
-        return latestPartitionRanges;
+        RangePartitionInfo rangePartitionInfo = (RangePartitionInfo) partitionTable.getPartitionInfo();
+        return rangePartitionInfo.getRangeList(filteredIds, false);
+    }
+
+    private List<Range<PartitionKey>> getLatestPartitionRange(Table table, Column partitionColumn,
+                                                              Set<String> modifiedPartitionNames) {
+        if (table.isLocalTable()) {
+            return getLatestPartitionRangeForLocalTable((OlapTable) table, modifiedPartitionNames);
+        } else {
+            Map<String, Range<PartitionKey>> partitionMap;
```

Comment:
here I think we should consider the properties of forceExternalTableQueryRewrite?

Reply:
forceExternalTableQueryRewrite is considered at prepareMvCandidatesForPlan, if forceExternalTableQueryRewrite is false, it will do not use MaterializedViewOptimizer to getLatestPartitionRange